core/src/main/java/com/radixdlt/RadixNodeApplication.java

```diff
@@ -110,22 +111,47 @@ public static void main(String[] args) {
       dumpExecutionLocation();
       // Bouncy Castle is required for loading the node key, so set it up now.
       setupBouncyCastle();
-
       final var properties = loadProperties(args);
-      final var radixNode = new RadixNode(properties);
-      radixNode.start();
-      Runtime.getRuntime().addShutdownHook(new Thread(radixNode::shutdown));
+      bootstrapRadixNode(properties);
     } catch (Exception ex) {
       log.fatal("Unable to start", ex);
       LogManager.shutdown(); // Flush any async logs
-
-      // This (or more likely, the one in ModuleRunnerImpl.java) may cause integration test errors
-      // which look like:
-      // "Process 'Gradle Test Executor 1' finished with non-zero exit value 255"
-      java.lang.System.exit(-1);
+      exitWithError();
     }
   }
 
+  private static void bootstrapRadixNode(RuntimeProperties properties) {
+    final var nodeBootStopwatch = Stopwatch.createStarted();
+    final var radixNodeBootstrapperHandle = RadixNodeBootstrapper.bootstrapRadixNode(properties);
+    // To clean up any resources if the application is shut down while the node is being
+    // bootstrapped
```

Comment:
I am afraid this will never be possible when we use the "obtain resource at construction" (ORAC) paradigm :(

What do I mean?
Consider this line:
https://github.com/radixdlt/babylon-node/blob/1f9e26b4a129014b95441d26ea037ef06cc47a23/core-rust-bridge/src/main/java/com/radixdlt/statemanager/StateManager.java#L92
It is executed during construction of `StateManager` (during injection, in our case). And it allocs rust native structures.
Let's imagine that guice continues with the object graph instantiation and stumbles upon some misconfig or a random problem, so it naturally throws. It never returned an instance of `StateManager` to you -> you have nothing to call the `.close()` upon.

This problem occurs every time when you obtain _any resource_ before your app driver (i.e. `main()`) captures a _reference_ to _the owner_ of _that resource_.
This can be sometimes hidden, e.g. here:
https://github.com/radixdlt/babylon-node/blob/0ebd2c1a0f513be5fc128455aa2fb41ba3857ed7/core/src/main/java/com/radixdlt/RadixNode.java#L136-L137
You may think: "the above is correct, because I first instantiate `systemApi` and then I start it"; however, a few other services are obtained and started from that same injector afterwards (which may throw), and the top-level `try/catch-all` does not receive the `systemApi` reference (not even its `injector`), and the system api server socket cannot be closed.

Another cool observation:
If our application followed the "do NOT obtain resources at construction" (DNORAC) paradigm, then the `addShutdownHook(radixNodeBootstrapperHandle::shutdown);` line would not be needed at all.
Why?
Because the bootstrapper does not hold any resources; its only responsibility is to construct/inject (and not `start()`) the `RadixNode`.

Our app is not very far from DNORAC, but making it 100% DNORAC is out of scope here I believe.
(please note, however, that failing to do DNORAC in a single class actually exposes all its dependencies to be resource leaks too; so it's not like we are 90% safe right now, because our top-level `RadixNode` cited above is ORAC).

To sum up:
- we can ditch the `RadixNodeBootstrapperHandle::shutdown`
  - it does not give us any guarantees with our current ORAC app code
  - it is not required when we do DNORAC
- we can simply return the `Future<>` from `.bootstrapRadixNode()`, saving some complexity
- if we care about cleaning up resources on spin-down, we can fix a few places and be DNORAC
  - actually, when I see that we create+destroy multiple (well, 2) injectors in a sequence, I am afraid that we have to care about that clean up :( (imagine that second injector cannot use some network port because the first one did not properly free it)
  - but on the other hand, one could argue that "well it only causes problems when there are exceptions and shit" and I would agree; but if we are close enough to having that right, I'd accept a minor time investment.

Reply:
_as discussed, I've added an explanatory comment and kept the `shutdown` call for now_